src/validators/model.rs

```diff
@@ -14,6 +14,7 @@ use crate::recursion_guard::RecursionGuard;
 use super::function::convert_err;
 use super::{build_validator, BuildValidator, CombinedValidator, Definitions, DefinitionsBuilder, Extra, Validator};
 
+const ROOT_FIELD: &str = "root";
```

Comment:
I know you already discussed it at length so I will try to ignore it in reviewing this PR, but I _hate_ doing this special casing for root models. Is there really now way to do this in Python?